### ISSUE TITLE
Strange preprocessor output

### ISSUE THREAD

**Issue by Azq2** [ShaderFrog/glsl-parser#31]
shader: https://github.com/libretro/glsl-shaders/blob/master/scalefx/shaders/scalefx-pass2.glsl

code
```js
	let fragment = preprocess(code, {
		defines: {
			FRAGMENT: true,
			//PARAMETER_UNIFORM: true,
			GL_ES: true,
			GL_FRAGMENT_PRECISION_HIGH: true,
		}
	});
```

result:
```
vec4 res;
res.x = min(jx.z + (1. - (jx.y)) * (1. - (jx.w)) * (1. - step(jSx.z, 0.)) * (jx.x + (1. - step(jSx.x + jSx.z, jSx.y + jSx.w))), 1.);
res.y = min(jy.w + (1. - (jy.z)) * (1. - (jy.x)) * (1. - step(jSy.w, 0.)) * (jy.y + (1. - step(jSy.jSy.x + jSy.z + jSy.w, jSy.x + jSy.z))), 1.);
res.z = min(jz.x + (1. - (jz.w)) * (1. - (jz.y)) * (1. - step(jSz.x, 0.)) * (jz.z + (1. - step(jSz.x + jSz.z, jSz.y + jSz.w))), 1.);
res.w = min(jw.y + (1. - (jw.x)) * (1. - (jw.z)) * (1. - step(jSw.0., 0.)) * (jw.w + (1. - step(jSw.jSw.x + jSw.z + jSw.w, jSw.x + jSw.z))), 1.);

```

expected:
```
vec4 res;
res.x = min(jx.z + (1. - (jx.y)) * (1. - (jx.w)) * (1. - step(jSx.z, 0.)) * (jx.x + (1. - step(jSx.x + jSx.z, jSx.y + jSx.w))), 1.);
res.y = min(jy.w + (1. - (jy.z)) * (1. - (jy.x)) * (1. - step(jSy.w, 0.)) * (jy.y + (1. - step(jSy.y + jSy.w, jSy.x + jSy.z))), 1.);
res.z = min(jz.x + (1. - (jz.w)) * (1. - (jz.y)) * (1. - step(jSz.x, 0.)) * (jz.z + (1. - step(jSz.x + jSz.z, jSz.y + jSz.w))), 1.);
res.w = min(jw.y + (1. - (jw.x)) * (1. - (jw.z)) * (1. - step(jSw.y, 0.)) * (jw.w + (1. - step(jSw.y + jSw.w, jSw.x + jSw.z))), 1.);
```

**Comment by AndrewRayCode** [comment]
I was able to boil down the issue to the most bare bones reproducible case, which is that if the source program has a macro replacement arg that's the same name as the macro arg, it triggers this issue. Created a failing test case for it: https://github.com/ShaderFrog/glsl-parser/pull/32/files

**Comment by AndrewRayCode** [comment]
Thanks for the bug report! Another bug I'm surprised wasn't caught for this long. Fixed in 5.1.0

```
npm install --save @shaderfrog/glsl-parser@5.1.0
```

**Comment by Azq2** [comment]
Thanks, bug fixed.
